dvc/repo/vdir/__init__.py

```diff
@@ -0,0 +1,29 @@
+from dataclasses import dataclass
+
+from dvc.hash_info import HashInfo
+
+
+class Vdir:
+    def __init__(self, repo):
+        self.repo = repo
+
+    def pull(self, *args, **kwargs):
+        from .pull import pull
+
+        return pull(self.repo, *args, **kwargs)
+
+    def cp(self, *args, **kwargs):
+        from .cp import cp
+
+        return cp(self.repo, *args, **kwargs)
+
+
+@dataclass
+class VirtualDirData:
+    operation: str  # e.g. cp, mv, rm
```

Comment:
Note that we already have things like `DvcTree` that have `copy/move/remove` implemented, maybe it would be better to use those somehow?

Reply:
Interesting! At the moment, the `VirtualDirData` is currently being injected inside the `BaseTree` so perhaps I think I can reuse the implementations in the `DvcTree`. Cheers for the heads up!